src/program_generator/pg_arythmetic.py

```diff
@@ -149,25 +158,20 @@ def _generate_random_arithmetic(self, gasLimit, opsLimit, bytecodeLimit, dominan
       operation = self._operations[op]
       if cleanStack:
         # the stack is empty, put one value there
-        bytecode += self._random_push(push)
+        bytecode += self._random_push(pushMax, randomizePush)
         ops_count += 1
         gas += 3
       # one value is always on the stack
       needed_pushes = int(operation['Removed from stack']) - 1
       # i.e. 23 from 0x23
       opcode = operation['Value'][2:4]
-      if op in arithmetic_ops or op in bitwise_ops or op in comparison_ops:
+      if op in arithmetic_ops or op in bitwise_ops or op in comparison_ops or op in exp_ops:
         for i in range(needed_pushes):
-          bytecode += self._random_push(push)
-      elif op in exp_ops:
-        bytecode += self._random_push(1)  # the exponent less than 256
-        bytecode += '90'  # SWAP1 so the exponent is first on the stack
-        ops_count += 1   # additional SWAP1 cost
-        gas += 3   # additional SWAP1 cost
```

Comment:
we don't need the `the exponent less than 256` restriction, so the swapping isn't necessary, EXP is done just like other binary ops

Reply:
Good point on the gas calculations though. Since we're not planning to rely on this, and it is clearly off for EXP now and I don't want to fix this now, I'll remove support for this in a follow-up commit. When we need it, we can add it back in.